google/cloud/pubsub/samples/samples.cc

```diff
@@ -221,6 +221,109 @@ void Subscribe(google::cloud::pubsub::Subscriber subscriber,
+    auto publisher = pubsub::Publisher(pubsub::MakePublisherConnection(
+        std::move(topic), pubsub::PublisherOptions{},
+        pubsub::ConnectionOptions{}.DisableBackgroundThreads(cq)));
+
+    std::vector<future<void>> ids;
+    for (char const* data : {"1", "2", "3", "go!"}) {
+      ids.push_back(
+          publisher.Publish(pubsub::MessageBuilder().SetData(data).Build())
+              .then([data](future<StatusOr<std::string>> f) {
+                auto s = f.get();
+                if (!s) return;
+                std::cout << "Sent '" << data << "' (" << *s << ")\n";
+              }));
+    }
+    publisher.Flush();
+    // Block until they are actually sent.
+    for (auto& id : ids) id.get();
+
+    // Shutdown the completion queue and join the threads
+    cq.Shutdown();
+    for (auto& t : tasks) t.join();
+  }
+  //! [custom-thread-pool-publisher]
+  (argv.at(0), argv.at(1));
+}
+
+void CustomThreadPoolSubscriber(std::vector<std::string> const& argv) {
+  namespace examples = ::google::cloud::testing_util;
+  if (argv.size() != 2 || (argv.size() == 1 && argv[0] == "--help")) {
+    throw examples::Usage{
+        "custom-thread-pool-subscriber <project-id> <topic-id>"};
+  }
+  //! [custom-thread-pool-subscriber]
+  namespace pubsub = google::cloud::pubsub;
+  using google::cloud::future;
+  using google::cloud::StatusOr;
+  [](std::string project_id, std::string subscription_id) {
+    // Create our own completion queue to run the background activity, such as
+    // flushing the publisher.
+    google::cloud::CompletionQueue cq;
+    // Setup 1 or more of threads to service this completion queue, these must
+    // remain running until all the work is done.
+    std::vector<std::thread> tasks;
+    std::generate_n(std::back_inserter(tasks), 4, [&cq] {
+      return std::thread(&google::cloud::CompletionQueue::Run, &cq);
+    });
+
+    auto subscriber = pubsub::Subscriber(pubsub::MakeSubscriberConnection(
+        pubsub::ConnectionOptions{}.DisableBackgroundThreads(cq)));
+    auto subscription =
+        pubsub::Subscription(std::move(project_id), std::move(subscription_id));
+
+    std::mutex mu;
+    int count = 0;
+    google::cloud::promise<void> received_message;
+    auto result = subscriber.Subscribe(
+        subscription, [&](pubsub::Message const& m, pubsub::AckHandler h) {
+          std::cout << "Received message " << m << std::endl;
+          {
+            std::lock_guard<std::mutex> lk(mu);
+            if (++count == 4) received_message.set_value();
+          }
+          std::move(h).ack();
+        });
+    // Cancel the subscription as soon as the first message is received.
+    received_message.get_future()
+        .then([&result](google::cloud::future<void>) { result.cancel(); })
+        .get();
+    // Report any final status, blocking until the subscription loop completes,
+    // either with a failure or because it is canceled.
```

Comment:
Opt: s/is canceled/was canceled/ perhaps?

Reply:
Done.